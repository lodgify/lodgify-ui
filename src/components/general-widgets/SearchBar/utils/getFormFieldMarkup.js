/* eslint react/prop-types: 0 */
import React, { Fragment } from 'react';
import { Form } from 'semantic-ui-react';
import { size } from 'lodash';

import { Icon, ICON_NAMES } from 'elements/Icon';
import { Dropdown } from 'inputs/Dropdown';
import { DateRangePicker } from 'inputs/DateRangePicker';

/**
 * @param  {Object}   props
 * @param  {string}   props.dateRangePickerLocaleCode
 * @param  {boolean}  props.isShowingSummary
 * @param  {Function} props.getIsDayBlocked
 * @param  {Object[]} props.locationOptions
 * @param  {Object[]} props.guestsOptions
 * @param  {Node}     props.searchButton
 * @param  {Function} onDatePickerChange
 * @param  {boolean}  areColumnsStacked
 * @param  {boolean}  willDropdownsOpenAbove
 * @return {Object}
 */
export const getFormFieldMarkup = (
  {
    dateRangePickerLocaleCode,
    isShowingSummary,
    getIsDayBlocked,
    locationOptions,
    guestsOptions,
    searchButton,
  },
  onDatePickerChange,
  areColumnsStacked,
  willDropdownsOpenAbove
) => {
  const defaultColumnWidth = areColumnsStacked ? 'twelve' : 'three';
  const datePickerColumnWidth = areColumnsStacked ? 'twelve' : 'seven';
  const buttonColumnWidth = areColumnsStacked ? 'twelve' : 'four';

  return (
    <Fragment>
      {!!isShowingSummary && (
        <Form.Field width={defaultColumnWidth}>
          <Icon
            isDisabled
            labelText="Property Summary"
            name={ICON_NAMES.HOME}
          />
        </Form.Field>
      )}
      {!!size(locationOptions) > 0 && (
        <Form.Field width={defaultColumnWidth}>
          <Dropdown
            icon={ICON_NAMES.MAP_PIN}
            label="Location"
            name="location"
            onChange={onDatePickerChange}
            options={locationOptions}
            willOpenAbove={willDropdownsOpenAbove}
          />
        </Form.Field>
      )}
      <Form.Field width={datePickerColumnWidth}>
        <DateRangePicker
          endDatePlaceholderText="Check-out"
          getIsDayBlocked={getIsDayBlocked}
          localeCode={dateRangePickerLocaleCode}
          name="dates"
          onChange={onDatePickerChange}
          startDatePlaceholderText="Check-in"
          willOpenAbove={willDropdownsOpenAbove}
        />
      </Form.Field>
      <Form.Field width={defaultColumnWidth}>
        <Dropdown
          icon={ICON_NAMES.USERS}
          label="Guests"
          name="guests"
          onChange={onDatePickerChange}
          options={guestsOptions}
          willOpenAbove={willDropdownsOpenAbove}
        />
      </Form.Field>
      <Form.Field width={buttonColumnWidth}>{searchButton}</Form.Field>
    </Fragment>
  );
};
