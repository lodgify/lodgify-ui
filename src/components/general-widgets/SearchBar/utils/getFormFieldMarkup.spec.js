import React from 'react';
import { shallow } from 'enzyme';

import { getFormFieldMarkup } from './getFormFieldMarkup';

const componentProps = {
  isShowingSummary: true,
  isShowingLocationDropdown: true,
  getIsDayBlocked: Function.prototype,
  locationOptions: [],
  guestsOptions: [],
  searchButton: <div />,
};

const willDropdownsOpenAbove = true;

const getMarkup = overrideProps =>
  shallow(
    <div>
      {getFormFieldMarkup(
        {
          ...componentProps,
          ...overrideProps,
        },
        Function.prototype,
        false,
        willDropdownsOpenAbove
      )}
    </div>
  );

describe('getFormFieldMarkup', () => {
  it('should render the right structure`', () => {
    const wrapper = getMarkup();

    expect(wrapper).toMatchSnapshot();
  });

  describe('if `props.locationOptions` size is greater than 0', () => {
    const getMerkupWithLocationOptions = () =>
      getMarkup({ locationOptions: [{ text: 'yo' }] });

    it('should render the right structure', () => {
      const wrapper = getMerkupWithLocationOptions();

      expect(wrapper).toMatchSnapshot();
    });
  });
});
