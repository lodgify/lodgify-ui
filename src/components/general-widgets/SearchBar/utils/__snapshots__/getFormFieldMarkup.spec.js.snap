// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFormFieldMarkup if \`props.locationOptions\` size is greater than 0 should render the right structure 1`] = `
<div>
  <FormField
    width="three"
  >
    <Icon
      color={null}
      hasBorder={false}
      isButton={false}
      isCircular={false}
      isColorInverted={false}
      isDisabled={true}
      isLabelLeft={false}
      labelText="Property Summary"
      labelWeight={null}
      name="home"
      path={null}
      size={null}
    />
  </FormField>
  <FormField
    width="three"
  >
    <Dropdown
      error={false}
      icon="map pin"
      isDisabled={false}
      isSearchable={false}
      isValid={false}
      label="Location"
      name="location"
      noResultsText="No results"
      onBlur={[Function]}
      onChange={[Function]}
      options={
        Array [
          Object {
            "text": "yo",
          },
        ]
      }
      willOpenAbove={false}
    />
  </FormField>
  <FormField
    width="seven"
  >
    <WithResponsive(DateRangePicker)
      endDatePlaceholderText="Check-out"
      getIsDayBlocked={[Function]}
      initialValue={true}
      name="dates"
      onChange={[Function]}
      startDatePlaceholderText="Check-in"
    />
  </FormField>
  <FormField
    width="three"
  >
    <Dropdown
      error={false}
      icon="users"
      isDisabled={false}
      isSearchable={false}
      isValid={false}
      label="Guests"
      name="guests"
      noResultsText="No results"
      onBlur={[Function]}
      onChange={[Function]}
      options={Array []}
      willOpenAbove={false}
    />
  </FormField>
  <FormField
    width="four"
  >
    <div />
  </FormField>
</div>
`;

exports[`getFormFieldMarkup should render the right structure\` 1`] = `
<div>
  <FormField
    width="three"
  >
    <Icon
      color={null}
      hasBorder={false}
      isButton={false}
      isCircular={false}
      isColorInverted={false}
      isDisabled={true}
      isLabelLeft={false}
      labelText="Property Summary"
      labelWeight={null}
      name="home"
      path={null}
      size={null}
    />
  </FormField>
  <FormField
    width="seven"
  >
    <WithResponsive(DateRangePicker)
      endDatePlaceholderText="Check-out"
      getIsDayBlocked={[Function]}
      initialValue={true}
      name="dates"
      onChange={[Function]}
      startDatePlaceholderText="Check-in"
    />
  </FormField>
  <FormField
    width="three"
  >
    <Dropdown
      error={false}
      icon="users"
      isDisabled={false}
      isSearchable={false}
      isValid={false}
      label="Guests"
      name="guests"
      noResultsText="No results"
      onBlur={[Function]}
      onChange={[Function]}
      options={Array []}
      willOpenAbove={false}
    />
  </FormField>
  <FormField
    width="four"
  >
    <div />
  </FormField>
</div>
`;
